build-bench-env.sh

```diff
@@ -61,7 +61,7 @@ readonly version_tcg=bb24fb0a8be3a6ef52888d247097d05976b8918e # 2023-04-22
 
 # benchmark versions
 readonly version_redis=6.2.7
-readonly version_lean=v3.4.2
+readonly version_lean=21d264a66d53b0a910178ae7d9529cb5886a39b6
```

Comment:
Could you leave a comment here about the commit ID rather than a version number, so future me knows why.

Reply:
Sure! I force-pushed to add one, I guess it is better to add other commits in case of numerous requested changes, will keep it in mind.